server/index.js

```diff
@@ -38,6 +38,28 @@ app.get('/api/products', (req, res, next) => {
     });
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  // { productId: '1'}
+  const id = req.params.productId;
+  const params = [id];
+  const sql = `
+      select *
+        from "products"
+        where "productId" = $1;
+      `;
+  db.query(sql, params)
+    .then(result => {
+      if (result.rows.length === 0) {
+        next(new ClientError(`Cannot find product with macthing id ${id}, 404`));
```

Comment:
`new ClientError()` should be passed 2 separate arguments, but here the `404` is inside your string.